### PR TITLE
Update oximeter dependency to pull in automatic producer registration

### PR DESCRIPTION
**Pull request by bnaecker**

### CODE REVIEW

**Comment by bnaecker** [comment]
Part of https://github.com/oxidecomputer/omicron/issues/5284.

File: bin/propolis-server/src/lib/server.rs [hunk 1]
```diff
@@ -343,136 +332,65 @@ fn instance_spec_from_request(
...
-                }
-                _ => omicron_common::backoff::BackoffError::permanent(
-                    e.to_string(),
-                ),
-            })
-        };
-        let warn_on_failure = |error, delay| {
-            warn!(
+    // The server manages all details of the registration with Nexus, so we
+    // don't need our own task for that or way to shut it down.
```
Comment (pfmooney) [inline bin/propolis-server/src/lib/server.rs:349]: That's convenient!

File: bin/propolis-server/src/lib/stats.rs [hunk 2]
```diff
@@ -169,13 +167,13 @@ pub async fn start_oximeter_server(
     );
     let config = Config {
         server_info,
-        registration_address,
-        dropshot: dropshot_config,
+        registration_address: Some(registration_address),
+        request_body_max_bytes: 2048,
```
Comment (pfmooney) [inline bin/propolis-server/src/lib/stats.rs:171]: Could we get some documentation for this magic number?  (And perhaps toss it in a `const`, while we're here?)

File: bin/propolis-server/src/lib/stats.rs [hunk 3]
```diff
@@ -132,33 +131,32 @@ impl Producer for ServerStatsOuter {
...
-    };
-
     let server_info = ProducerEndpoint {
         id,
         kind: ProducerKind::Instance,
-        address: my_address,
-        base_route: "/collect".to_string(),
+        address: producer_address,
+        // NOTE: This field is unused, and will be removed in the future.
+        base_route: String::new(),
```
Comment (hawkw) [inline bin/propolis-server/src/lib/stats.rs:159]: tiny nit, not important: is there an Oximeter issue or documentation or something that this comment could include a link to?

File: bin/propolis-server/src/lib/stats.rs (hunk 2 above)
Comment (hawkw) [reply bin/propolis-server/src/lib/stats.rs:171]: FWIW, it looks like it was also a magic number previously, and this commit just moved it: https://github.com/oxidecomputer/propolis/pull/689/files#diff-b0ffb61cea8161225fe6897fd730efc8c2521123536783a990abe00d37fccbb6L152

But, I agree that a constant would be nice.

File: bin/propolis-server/src/lib/stats.rs (hunk 3 above)
Comment (bnaecker) [reply bin/propolis-server/src/lib/stats.rs:159]: Just made https://github.com/oxidecomputer/omicron/issues/5658 to track, will link here.

File: bin/propolis-server/src/lib/server.rs [hunk 4]
```diff
@@ -238,14 +228,14 @@ impl ServiceProviders {
...
-            oximeter_state.registration_task.take()
-        {
-            let _ = shutdown_tx.send(());
-            task.abort();
-        };
-        if let Some(server) = oximeter_state.server_task.take() {
-            server.abort();
+        if let Some(server) = oximeter_state.server.take() {
+            if let Err(e) = server.close().await {
+                slog::error!(
```
[the remark this answers is not in the document]
Comment (bnaecker) [reply bin/propolis-server/src/lib/server.rs]: Done in 32b270c

File: bin/propolis-server/src/lib/stats.rs (hunk 3 above)
Comment (hawkw) [reply bin/propolis-server/src/lib/stats.rs]: lovely, thank you!

File: bin/propolis-server/src/lib/stats.rs [hunk 5]
```diff
@@ -167,15 +166,22 @@ pub async fn start_oximeter_server(
     let producer_log = oximeter_producer::LogConfig::Logger(
         log.new(slog::o!("component" => "oximeter-producer")),
     );
+
+    // The maximum size of a single Dropshot request.
+    //
+    // This is a pretty arbitrary limit, but one that should be big enough for
+    // the statistics we serve today (vCPU usage and panic counts), with
+    // headroom for adding quite a few more.
+    const MAX_REQUEST_SIZE: usize = 1024 * 1024;
```
Comment (hawkw) [inline bin/propolis-server/src/lib/stats.rs:175]: looks like we also made this a lot bigger --- it was 2 KiB before, and now it's 1 MiB. was that intentional?

**Review by hawkw — APPROVED** [review]
overall, this looks good. i noticed that we've increased the maximum request size substantially --- did we mean to do that?

**Comment by bnaecker** [comment]
Thanks for the review y'all. I'm waiting to land https://github.com/oxidecomputer/omicron/pull/5659 and then pull that in here, so that this can always be run against the `oximeter-standalone` binary for testing.

**Pull request merged**